action.yml

```
@@ -0,0 +1,186 @@
+name: "End of Life and Outdated JS Scan"
+description: "Reusable composite action for scanning dependencies for end-of-life and outdated JS packages."
+
+inputs:
+  github-token:
+    description: "GitHub token for creating/updating comments"
+    required: true
+  project:
+    description: "Project name for dependency-check"
+    required: true
+  scan:
+    description: "Path to run dependency-check on"
+    required: false
+    default: "."
+  suppression:
+    description: "Path to the suppression.xml file"
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set suppression argument
+      id: suppression_arg
+      shell: bash
+      run: |
+        if [ -z "${{ inputs.suppression_file }}" ]; then
+          echo "suppression=" >> $GITHUB_OUTPUT
+        else
+          echo "suppression=--suppression ${{ inputs.suppression_file }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Run Dependency Check
+      uses: dependency-check/Dependency-Check_Action@main
+      with:
+        path: ${{ inputs.path }}
+        format: 'JSON'
+        project: ${{ inputs.project }}
+        args: >
+          --failOnCVSS 7
+          --exclude "**/node_modules/**"
+          ${{ steps.suppression_arg.outputs.suppression }}
+
+    - name: Generate comment
+      if: ${{ !cancelled() }}
+      id: generate_comment
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const fs = require('fs');
+
+          const filePath = 'reports/dependency-check-report.json';
+          if (!fs.existsSync(filePath)) {
+            console.log('No dependency-check JSON report found.');
+            fs.appendFileSync(process.env.GITHUB_OUTPUT, `vuln_count=0\ncomment<<EOF\n\nEOF\n`);
+            process.exit(0);
+          }
```

The action will fail, right?

---

No, this would pass if no report was generated. But thinking, i think we should fail and show error message saying report was not found. Changed this